### PR TITLE
feat: Remove sales channels from pub keys

### PR DESCRIPTION
**Pull request by olivermrbl**
**What**
- Add workflow + step for detaching sales channels from pub API keys
- Tweak linking error message to be more helpful
- Add `removeRemoteLink` step to delete API key workflow

### CODE REVIEW

File: packages/core-flows/src/api-key/workflows/delete-api-keys.ts [hunk 1]
```diff
@@ -7,6 +9,10 @@ export const deleteApiKeysWorkflowId = "delete-api-keys"
 export const deleteApiKeysWorkflow = createWorkflow(
   deleteApiKeysWorkflowId,
   (input: WorkflowData<WorkflowInput>): WorkflowData<void> => {
-    return deleteApiKeysStep(input.ids)
+    deleteApiKeysStep(input.ids)
+
+    removeRemoteLinkStep({
```
Comment (sradevski) [inline packages/core-flows/src/api-key/workflows/delete-api-keys.ts:15]: NIT: Might be good to mention that the API keys might not be `publishable` keys (although there is no harm in running this command). Alternatively we can fetch before deleting, but I think it's not really necessary

File: packages/core-flows/src/api-key/steps/detach-sales-channels-from-publishable-keys.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    sales_channel_ids: string[]
+  }[]
+}
+
+export const detachApiKeysWithSalesChannelsStepId =
+  "detach-sales-channels-with-api-keys"
+export const detachApiKeysWithSalesChannelsStep = createStep(
+  detachApiKeysWithSalesChannelsStepId,
+  async (input: StepInput, { container }) => {
+    const remoteLink = container.resolve(ContainerRegistrationKeys.REMOTE_LINK)
```
Comment (sradevski) [inline packages/core-flows/src/api-key/steps/detach-sales-channels-from-publishable-keys.ts:17]: Let's check the input length before trying to create a link

**Review by sradevski — APPROVED** [review]

File: packages/modules-sdk/src/utils/linking-error.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  moduleB: string
+  moduleBKey: string
+  type: "dismiss" | "link"
+}
+
+/**
+ *
+ * Example: Module to dismiss salesChannel and apiKey by keys sales_channel_id and api_key_id was not found. Ensure the link exists, keys are correct, and the link is passed in the correct order to method 'remoteLink.dismiss'
+ */
+export const linkingErrorMessage = (input: LinkingErrorMessageInput) => {
```
Comment (olivermrbl) [inline packages/modules-sdk/src/utils/linking-error.ts:18]: @carlos-r-l-rodrigues, this change OK for you?

File: packages/core-flows/src/api-key/workflows/remove-sales-channels-from-publishable-key.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+type WorkflowInput = {
+  data: {
+    api_key_id: string
+    sales_channel_ids: string[]
+  }[]
+}
+
+export const removeSalesChannelsFromApiKeyWorkflowId =
+  "remove-sales-channels-From-api-key"
```
Comment (fPolic) [inline packages/core-flows/src/api-key/workflows/remove-sales-channels-from-publishable-key.ts]:
```suggestion
  "remove-sales-channels-from-api-key"
```

**Review by fPolic — APPROVED** [review]

File: packages/modules-sdk/src/utils/linking-error.ts (hunk 3 above)
Comment (olivermrbl) [reply packages/modules-sdk/src/utils/linking-error.ts:18]: Just bumping this one in case you missed it 😄

Comment (carlos-r-l-rodrigues) [reply packages/modules-sdk/src/utils/linking-error.ts:18]: Absolutely...

**Pull request merged**